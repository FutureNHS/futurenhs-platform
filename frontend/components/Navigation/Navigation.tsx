import React from "react";

import styled from "styled-components";
import { v4 as uuid } from "uuid";

import { NavListItem } from "../NavListItem";

interface Workspace {
  title: string;
  // folders: any[]; //TODO!
}
interface Props {
  workspace: Workspace;
}

const Nav = styled.nav`
  border: 1px solid red;
  width: 270px;
  ${({ theme }) => `
  @media (min-width: ${theme.mqBreakpoints.tablet}) {
      width: 300px;
    }

  @media (max-width: ${theme.mqBreakpoints.largeDesktop}) {
      width: 360px;
    }
  `}
`;

const folders = [{ title: "Folder 1" }, { title: "Folder 2" }];
const Navigation = ({ workspace }: Props) => (
  <Nav>
    <div>
      <h3>{workspace.title}</h3>
      <a>About this workspace</a>
    </div>
    <div>
      <section>
        <h4>Quick Links</h4>
      </section>
      <section>
        <h4>Folders</h4>
        <ul>
          {folders.map((folder) => (
            // <li key={folder.title}>{folder.title}</li>
            <NavListItem key={uuid()} folder={folder} />
          ))}
        </ul>
      </section>
    </div>
  </Nav>
);

export default Navigation;
