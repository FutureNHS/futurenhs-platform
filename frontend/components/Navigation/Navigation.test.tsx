import React from "react";

import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import theme from "../../lib/fixtures/theme.json";
import Navigation from "./Navigation";

describe("<Navigation/>", () => {
  const folders = [
    { id: "1234", title: "Folder 1" },
    { id: "5678", title: "Folder 2" },
  ];
  const workspace = { id: "1111", title: "My workspace" };
  it("renders correctly", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <Navigation workspace={workspace} folders={folders} />
      </ThemeProvider>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
