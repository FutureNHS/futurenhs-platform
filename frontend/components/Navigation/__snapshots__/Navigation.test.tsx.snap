// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> renders correctly 1`] = `
<DocumentFragment>
  .c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 100%;
  border-radius: 4px;
  margin-left: 7px;
}

.c7 .tooltip {
  left: 36px;
  top: -7px;
}

.c7 .folder-menu-item {
  display: none;
}

.c7 .folder-menu-item a {
  color: rgb(255,255,255);
}

.c7 .folder-menu-item a:focus {
  color: rgb(33,43,50);
}

.c9 {
  display: inline-block;
  position: relative;
}

.c12 {
  opacity: 0;
  position: absolute;
  z-index: 2000;
  border-radius: 4px;
  background-color: rgb(76,98,114);
  padding: 5px;
  font-weight: 400;
  font-size: 18px;
  color: rgb(255,255,255);
}

.c8:hover .c11 {
  opacity: 1;
  -webkit-transition: opacity 0s linear 0.5s;
  transition: opacity 0s linear 0.5s;
}

.c12::after {
  position: absolute;
  bottom: 33%;
  left: -10%;
  margin-left: -2px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent rgb(76,98,114) transparent transparent;
  content: " ";
}

.c10 {
  opacity: 1;
  color: rgb(76,98,114);
  background-color: rgb(216,221,224);
  border-radius: 4px;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  height: 100%;
  line-height: 24px;
  padding: 0;
  position: relative;
}

.c10.hidden {
  opacity: 0;
}

.c10:hover {
  opacity: 1;
  color: rgb(255,255,255);
  background-color: rgb(0,61,120);
}

.c10:hover.hidden {
  color: rgb(255,255,255);
}

.c10.open,
.c10:active,
.c10:focus {
  opacity: 1;
  color: rgb(33,43,50);
  background-color: rgb(255,235,59);
}

.c10.open.hidden,
.c10:active.hidden,
.c10:focus.hidden {
  color: rgb(33,43,50);
}

.c10 .icon-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  list-style-type: none;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5:hover > div {
  background-color: rgb(216,221,224);
}

.c5:hover > div button {
  opacity: 1;
  color: rgb(76,98,114);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: inherit;
  border-radius: 4px;
  width: 227px;
  margin-right: 4px;
  font-weight: inherit;
}

.c6 a {
  height: 100%;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c6 a:focus {
  box-shadow: none;
}

.c6 img {
  padding-top: 2px;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c6 div {
  padding-left: 4px;
  padding-right: 4px;
  color: rgb(33,43,50);
}

.c3 {
  padding-top: 20px;
}

.c3 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c3 button {
  border: none;
  background: inherit;
}

.c3 h4 {
  color: rgb(0,94,184);
}

.c0 {
  display: none;
  padding-top: 24px;
  padding-right: 52px;
  padding-left: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  box-sizing: border-box;
  width: 270px;
}

.c1 {
  padding-bottom: 20px;
  border-bottom: 1px solid rgb(76,98,114);
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c2:hover {
  color: inherit;
}

.c2:visited {
  color: inherit;
}

.c4 {
  padding-left: 0px;
}

.c4 .nav-list-item div {
  font-weight: 700;
  color: rgb(0,94,184);
}

@media (min-width:990px) {
  .c7 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .c7 .folder-menu-item {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:641px) {
  .c0 {
    display: block;
    width: 300px;
  }
}

@media (min-width:990px) {
  .c0 {
    width: 360px;
  }
}

<nav
    class="c0"
  >
    <header
      class="c1"
    >
      <a
        class="c2"
        href="/workspaces/1111"
      >
        <h3>
          My workspace
        </h3>
      </a>
      <a
        href="/workspaces/1111"
      >
        About this workspace
      </a>
    </header>
    <section
      class="c3"
    >
      <div>
        <h4>
          Folders
        </h4>
        <button>
          <img
            alt="Hide Folders"
            src="/public/chevronOpen.svg"
          />
        </button>
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5 nav-list-item"
        >
          <div
            class="c6"
          >
            <a
              href="/workspaces/1111/folders/create-folder"
            >
              <img
                alt=""
                src="/public/createFolder.svg"
              />
              <div>
                Create new folder
              </div>
            </a>
          </div>
        </li>
        <li
          class="c5"
        >
          <div
            class="c6"
          >
            <a
              href="/workspaces/1111/folders/1234"
            >
              <img
                alt="folder"
                src="/public/folderClosed.svg"
              />
              <div>
                Folder 1
              </div>
            </a>
          </div>
          <div
            class="c7"
          >
            <div
              class="c8 c9"
            >
              <button
                aria-label="Options"
                class="hidden c10"
              >
                <div
                  class="icon-wrapper "
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M17 12a2 2 0 104 0 2 2 0 00-4 0zm-5 2a2 2 0 110-4 2 2 0 010 4zm-7 0a2 2 0 110-4 2 2 0 010 4z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </button>
              <span
                class="c11 c12 tooltip"
              >
                Options
              </span>
            </div>
          </div>
        </li>
        <li
          class="c5"
        >
          <div
            class="c6"
          >
            <a
              href="/workspaces/1111/folders/5678"
            >
              <img
                alt="folder"
                src="/public/folderClosed.svg"
              />
              <div>
                Folder 2
              </div>
            </a>
          </div>
          <div
            class="c7"
          >
            <div
              class="c8 c9"
            >
              <button
                aria-label="Options"
                class="hidden c10"
              >
                <div
                  class="icon-wrapper "
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M17 12a2 2 0 104 0 2 2 0 00-4 0zm-5 2a2 2 0 110-4 2 2 0 010 4zm-7 0a2 2 0 110-4 2 2 0 010 4z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </button>
              <span
                class="c11 c12 tooltip"
              >
                Options
              </span>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </nav>
</DocumentFragment>
`;
